pkg/interfaces/contracts/vault/IAggregatorCompositeLiquidityRouter.sol

```diff
@@ -0,0 +1,139 @@
+ * @dev This contract allow interacting with ERC4626 Pools (which contain wrapped ERC4626 tokens) using only underlying
+ * standard tokens. For instance, with `addLiquidityUnbalancedToERC4626Pool` it is possible to add liquidity to an
+ * ERC4626 Pool with [waDAI, waUSDC], using only DAI, only USDC, or an arbitrary amount of both. If the ERC4626 buffers
+ * in the Vault have liquidity, these will be used to avoid wrapping/unwrapping through the wrapped token interface,
+ * saving gas.
+ *
+ * For instance, adding only DAI to the pool above (and assuming an aDAI buffer with enough liquidity), would pull in
+ * the DAI from the user, swap it for waDAI in the internal Vault buffer, and deposit the waDAI into the ERC4626 pool:
+ * 1) without having to do any expensive ERC4626 wrapping operations; and
+ * 2) without requiring the user to construct a batch operation containing the buffer swap.
+ *
+ * The aggregator composite liquidity router uses prepayment instead of permit2.
+ */
+interface IAggregatorCompositeLiquidityRouter is ICompositeLiquidityRouterQueries {
+    /***************************************************************************
+                                   ERC4626 Pools
+    ***************************************************************************/
+
+    /**
+     * @notice Add arbitrary amounts of tokens to an ERC4626 pool through the buffer.
+     * @dev An "ERC4626 pool" contains IERC4626 yield-bearing tokens (e.g., waDAI). Ensure that any buffers associated
+     * with the wrapped tokens in the ERC4626 pool have been initialized before initializing or adding liquidity to
+     * the "parent" pool, and also make sure limits are set properly.
+     *
+     * @param pool Address of the liquidity pool
+     * @param wrapUnderlying Flags indicating whether the corresponding token should be wrapped or used as an ERC20
+     * @param exactAmountsIn Exact amounts of underlying/wrapped tokens in, sorted in token registration order
+     * @param minBptAmountOut Minimum amount of pool tokens to be received
+     * @param userData Additional (optional) data required for adding liquidity
+     * @return bptAmountOut Actual amount of pool tokens received
+     */
+    function addLiquidityUnbalancedToERC4626Pool(
+        address pool,
+        bool[] memory wrapUnderlying,
+        uint256[] memory exactAmountsIn,
+        uint256 minBptAmountOut,
+        bytes memory userData
+    ) external payable returns (uint256 bptAmountOut);
+
+    /**
+     * @notice Add proportional amounts of tokens to an ERC4626 pool through the buffer.
+     * @dev An "ERC4626 pool" contains IERC4626 yield-bearing tokens (e.g., waDAI). Ensure that any buffers associated
+     * with the wrapped tokens in the ERC4626 pool have been initialized before initializing or adding liquidity to
+     * the "parent" pool, and also make sure limits are set properly.
```

Comment:
Worth documenting that this function allows partial boosted pools. I'd also remove the "before initializing or", since it's weird to read "have been initialized before initializing". I assume an aggregator will not initialize pools, so the initialization process of a pool is irrelevant, and talking only about add liquidity here is simpler.

```suggestion
     * @dev An "ERC4626 pool" contains at least one IERC4626 yield-bearing token (e.g., waDAI). Ensure that any buffers associated
     * with the wrapped tokens in the ERC4626 pool have been initialized before adding liquidity to
     * the "ERC4626 pool", and also make sure limits are set properly.
```